lib/new_post_manager.rb

```diff
@@ -82,7 +82,18 @@ def self.post_needs_approval?(manager)
     is_fast_typer?(manager) ||
     matches_auto_silence_regex?(manager) ||
     WordWatcher.new("#{manager.args[:title]} #{manager.args[:raw]}").requires_approval? ||
-    (SiteSetting.approve_unless_staged && user.staged)
+    (SiteSetting.approve_unless_staged && user.staged) ||
+    post_needs_approval_in_its_category?(manager)
+  end
+
+  def self.post_needs_approval_in_its_category?(manager)
+    if manager.args[:topic_id].present?
+      Category.joins(:topics).find_by(topics: { id: manager.args[:topic_id] }).require_reply_approval?
```

Comment:
Oh ... what if the topic_id is invalid here or category_id on topic is null (like messages) `.require_reply_approval?` will explode...